lib/bridge/IrcBridge.js

```diff
@@ -897,6 +906,26 @@ IrcBridge.prototype.getBotClient = function(server) {
     });
 }
 
+IrcBridge.prototype._fetchJoinedRooms = Promise.coroutine(function*() {
+    const bot = this._bridge.getBot();
+    let gotRooms = false;
+    while (!gotRooms) {
+        try {
+            const roomids = yield bot.getJoinedRooms();
+            gotRooms = true;
+            this.joinedRoomList = roomids;
+            log.info(`ASBot is in ${roomids.length} rooms!`);
+        }
+        catch (ex) {
+            // We should retry this, because it's still quicker than rejoining everything.
```

Comment:
*Quicker to fail and succeed once you mean? The caveat with this is that it could block the bridge from ever starting.

What would be the typical circumstances for failing here and does it make sense to retry? Or block the bridge starting on retrying? (If it fails to request the joined rooms then that implies it won't be able to do much else.)